### DISCUSSION TITLE
Need Help with skip checkup failure

### DISCUSSION THREAD

**Discussion by mdaymard1** [skiptools/skip#300] [Q&A] [answered]
Hi,

I'm getting an error when running the skip check command, and would really appreciate a little guidance. Looks like it's related to to git:

macbook-pro-1 ~ % skip checkup --no-fail-fast
[✓] Skip version 1.2.8 (= 1.2.8)
[✓] macOS version 14.7 (> 13.5.0)
[✓] macOS architecture: ARM
[✓] Swift version 5.10 (> 5.9.0)
[✓] Xcode version 15.4 (> 15.0.0)
[✓] Xcode tools SDKs: 8
[✓] Homebrew version 4.4.11 (> 4.1.0)
[✓] Gradle version 8.11.1 (> 8.6.0)
[✓] Java version 23.0.1 (> 21.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2024.2
[✓] Android tools SDKs: 2
[✗] Resolve dependencies (0.91s)
[✗] Build hello-skip (0.25s)
[✗] Test Swift (0.25s)
[✗] Test Kotlin (0.25s)
[✗] Archive iOS ipa (2.17s)
[✗] The file “Payload” doesn’t exist.
[✗] Skip 1.2.8 checkup failed with 6 errors

here is the log after running with --verbose:

executing command: /opt/homebrew/bin/skip version
Skip version 1.2.8
[✓] Skip version 1.2.8 (= 1.2.8)
executing command: /usr/bin/sw_vers --productVersion
14.7
[✓] macOS version 14.7 (> 13.5.0)
executing command: /usr/sbin/sysctl -n sysctl.proc_translated
0
[✓] macOS architecture: ARM
executing command: /usr/bin/swift -version
Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
swift-driver version: 1.90.11.1 
[✓] Swift version 5.10 (> 5.9.0)
executing command: /usr/bin/xcodebuild -version
Xcode 15.4
Build version 15F31d
[✓] Xcode version 15.4 (> 15.0.0)
[✓] Xcode tools SDKs: 8
executing command: /opt/homebrew/bin/brew --version
Homebrew 4.4.11
[✓] Homebrew version 4.4.11 (> 4.1.0)
executing command: /opt/homebrew/bin/gradle -version
------------------------------------------------------------
Gradle 8.11.1
------------------------------------------------------------
Build time:    2024-11-20 16:56:46 UTC
Revision:      481cb05a490e0ef9f8620f7873b83bd8a72e7c39
Kotlin:        2.0.20
Groovy:        3.0.22
Ant:           Apache Ant(TM) version 1.10.14 compiled on August 16 2023
Launcher JVM:  23.0.1 (Homebrew 23.0.1)
Daemon JVM:    /opt/homebrew/Cellar/openjdk/23.0.1/libexec/openjdk.jdk/Contents/Home (no JDK specified, using current Java home)
OS:            Mac OS X 14.7 aarch64
[✓] Gradle version 8.11.1 (> 8.6.0)
executing command: /opt/homebrew/Cellar/openjdk/23.0.1/libexec/openjdk.jdk/Contents/Home/bin/java -version
openjdk version "23.0.1" 2024-10-15
OpenJDK Runtime Environment Homebrew (build 23.0.1)
OpenJDK 64-Bit Server VM Homebrew (build 23.0.1, mixed mode, sharing)
[✓] Java version 23.0.1 (> 21.0.0)
executing command: /opt/homebrew/bin/adb version
Android Debug Bridge version 1.0.41
Version 35.0.2-12147458
Installed as /opt/homebrew/bin/adb
Running on Darwin 23.6.0 (arm64)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2024.2
[✓] Android tools SDKs: 2
executing command: /usr/bin/swift package resolve -v --package-path /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T//DF71DD92-8159-454A-8DCE-F0619BC27A5E/hello-skip
warning: 'hello-skip': /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -c -primary-file /private/var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/DF71DD92-8159-454A-8DCE-F0619BC27A5E/hello-skip/Package.swift -target arm64-apple-macosx13.0 -Xllvm -aarch64-use-tbi -enable-objc-interop -stack-check -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -I /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib -I /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI -F /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks -vfsoverlay /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/TemporaryDirectory.t3h3ws/vfs.yaml -swift-version 5 -package-description-version 5.9.0 -new-driver-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-driver -disable-implicit-concurrency-module-import -disable-implicit-string-processing-module-import -empty-abi-descriptor -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -module-name main -disable-clang-spi -target-sdk-version 14.5 -target-sdk-name macosx14.5 -external-plugin-path '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/bin/swift-plugin-server' -external-plugin-path '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/local/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/bin/swift-plugin-server' -external-plugin-path '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/bin/swift-plugin-server' -external-plugin-path '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/local/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/bin/swift-plugin-server' -plugin-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/host/plugins -plugin-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/local/lib/swift/host/plugins -o /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/TemporaryDirectory.tXRTDk/Package-1.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/TemporaryDirectory.tXRTDk/Package-1.o -F /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks --sysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk --target=arm64-apple-macosx13.0 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx/libswiftCompatibilityPacks.a -L /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/lib/swift -L /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI -L /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib -lPackageDescription -Xlinker -rpath -Xlinker /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI -o /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/TemporaryDirectory.7o5LRC/hello-skip-manifest
Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx13.0
Running resolver because the following dependencies were added: 'skip' (https://source.skip.tools/skip.git), 'skip-ui' (https://source.skip.tools/skip-ui.git)
Fetching https://source.skip.tools/skip.git from cache
Fetching https://source.skip.tools/skip-ui.git from cache
Fetching https://source.skip.tools/skip.git from cache
Fetching https://source.skip.tools/skip-ui.git from cache
error: GitShellError(result: <ProcessResult: exit: terminated(code: 1), output:
 
Thanks in advance!
Mike

**Comment by YaroslavErmolaev** [comment]
My solution for this situation is to use gradle exactly the version specified: Gradle 8.6.0

**Reply by mdaymard1** [reply]
Hi,

Thanks for your suggestion. I downgraded Gradle to 8.6, but am still getting the same error.

FYI: I am running on Sonoma 14.7 and have XCode 15.4 installed

executing command: /opt/homebrew/bin/skip version
Skip version 1.2.8
[✓] Skip version 1.2.8 (= 1.2.8)
executing command: /usr/bin/sw_vers --productVersion
14.7
[✓] macOS version 14.7 (> 13.5.0)
executing command: /usr/sbin/sysctl -n sysctl.proc_translated
0
[✓] macOS architecture: ARM
executing command: /usr/bin/swift -version
Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
swift-driver version: 1.90.11.1 
[✓] Swift version 5.10 (> 5.9.0)
executing command: /usr/bin/xcodebuild -version
Xcode 15.4
Build version 15F31d
[✓] Xcode version 15.4 (> 15.0.0)
[✓] Xcode tools SDKs: 8
executing command: /opt/homebrew/bin/brew --version
Homebrew 4.4.11
[✓] Homebrew version 4.4.11 (> 4.1.0)
executing command: /Users/mikeaymard/.sdkman/candidates/gradle/current/bin/gradle -version
Welcome to Gradle 8.6!
Here are the highlights of this release:
 - Configurable encryption key for configuration cache
 - Build init improvements
 - Build authoring improvements
For more details see https://docs.gradle.org/8.6/release-notes.html
------------------------------------------------------------
Gradle 8.6
------------------------------------------------------------
Build time:   2024-02-02 16:47:16 UTC
Revision:     d55c486870a0dc6f6278f53d21381396d0741c6e
Kotlin:       1.9.20
Groovy:       3.0.17
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          23.0.1 (Homebrew 23.0.1)
OS:           Mac OS X 14.7 aarch64
[✓] Gradle version 8.6 (= 8.6.0)
executing command: /opt/homebrew/Cellar/openjdk/23.0.1/libexec/openjdk.jdk/Contents/Home/bin/java -version
openjdk version "23.0.1" 2024-10-15
OpenJDK Runtime Environment Homebrew (build 23.0.1)
OpenJDK 64-Bit Server VM Homebrew (build 23.0.1, mixed mode, sharing)
[✓] Java version 23.0.1 (> 21.0.0)
executing command: /opt/homebrew/bin/adb version
Android Debug Bridge version 1.0.41
Version 35.0.2-12147458
Installed as /opt/homebrew/bin/adb
Running on Darwin 23.6.0 (arm64)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2024.2
[✓] Android tools SDKs: 2
executing command: /usr/bin/swift package resolve -v --package-path /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T//84F550B0-035D-462F-968A-3044052F52E7/hello-skip
warning: 'hello-skip': /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -c -primary-file /private/var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/84F550B0-035D-462F-968A-3044052F52E7/hello-skip/Package.swift -target arm64-apple-macosx13.0 -Xllvm -aarch64-use-tbi -enable-objc-interop -stack-check -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -I /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib -I /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI -F /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks -vfsoverlay /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/TemporaryDirectory.JDUarX/vfs.yaml -swift-version 5 -package-description-version 5.9.0 -new-driver-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-driver -disable-implicit-concurrency-module-import -disable-implicit-string-processing-module-import -empty-abi-descriptor -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -module-name main -disable-clang-spi -target-sdk-version 14.5 -target-sdk-name macosx14.5 -external-plugin-path '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/bin/swift-plugin-server' -external-plugin-path '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/local/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/bin/swift-plugin-server' -external-plugin-path '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/bin/swift-plugin-server' -external-plugin-path '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/local/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/bin/swift-plugin-server' -plugin-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/host/plugins -plugin-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/local/lib/swift/host/plugins -o /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/TemporaryDirectory.n89pFd/Package-1.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/TemporaryDirectory.n89pFd/Package-1.o -F /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks --sysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk --target=arm64-apple-macosx13.0 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx/libswiftCompatibilityPacks.a -L /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/lib/swift -L /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI -L /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib -lPackageDescription -Xlinker -rpath -Xlinker /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI -o /var/folders/y_/ck4y4pln4y90g4mg8bh04lv80000gn/T/TemporaryDirectory.vimSNp/hello-skip-manifest
Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx13.0
Running resolver because the following dependencies were added: 'skip' (https://source.skip.tools/skip.git), 'skip-ui' (https://source.skip.tools/skip-ui.git)
Fetching https://source.skip.tools/skip-ui.git from cache
Fetching https://source.skip.tools/skip.git from cache
Fetching https://source.skip.tools/skip.git from cache
Fetching https://source.skip.tools/skip-ui.git from cache
error: GitShellError(result: <ProcessResult: exit: terminated(code: 1), output:
 
>)
[✗] Resolve dependencies (4.95s)
[✗] error: GitShellError(result: <ProcessResult: exit: terminated(code: 1), output:
[✗] Skip 1.2.8 checkup failed with 2 errors

**Reply by mdaymard1** [reply, marked as answer]
ok, the problem had to do with some extra metadata that SourceTree was inserting into my .gitconfig file. After removing that it ran fine:

[✓] Skip version 1.2.8 (= 1.2.8)
[✓] macOS version 14.7 (> 13.5.0)
[✓] macOS architecture: ARM
[✓] Swift version 5.10 (> 5.9.0)
[✓] Xcode version 15.4 (> 15.0.0)
[✓] Xcode tools SDKs: 8
[✓] Homebrew version 4.4.11 (> 4.1.0)
[✓] Gradle version 8.11.1 (> 8.6.0)
[✓] Java version 23.0.1 (> 21.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2024.2
[✓] Android tools SDKs: 2
[✓] Resolve dependencies (9.64s)
[✓] Build hello-skip (9.76s)
[✓] Test Swift (5.81s)
[✓] Test Kotlin (129.61s)
[✓] Archive iOS ipa (17.06s)
[✓] Assemble HelloSkip-release.ipa 63 KB
[✓] Verify HelloSkip-release.ipa 63 KB
[✓] Assembling Android apk (111.53s)
[✓] Verify HelloSkip-release.apk 14.8 MB
[✓] Check Swift Package (0.26s)
[✓] Check Skip Updates: 1.2.8
[✓] Skip 1.2.8 checkup succeeded in 285.19s

Thanks,
Mike
